### PR TITLE
Add possibility to attach user to checkout by App or staff user

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it allows attaching customer to checkout by external app or staff user

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/checkout/mutations.py [hunk 1]
```diff
@@ -612,6 +613,13 @@ class Arguments:
                 "DEPRECATED: Will be removed in Saleor 4.0. Use token instead."
             ),
         )
+        customer_id = graphene.ID(
+            required=False,
+            description=(
+                "ID of customer to attach to checkout. Can be used to attach customer "
+                "to checkout by staff or app. Requires manage_users permission."
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py:620]: In schema and in API reference permissions are capitalized enum values, so:
```suggestion
                "to checkout by staff or app. Requires MANAGE_USERS permission."
```

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/checkout/mutations.py [hunk 2]
```diff
@@ -1312,14 +1332,21 @@ def perform_mutation(
             checkout_info = fetch_checkout_info(
                 checkout, lines, info.context.discounts, manager
             )
+
+            requestor = get_user_or_app_from_context(info.context)
+            if requestor.has_perm(AccountPermissions.IMPERSONATE_USER):
+                customer = checkout.user
+            else:
+                customer = info.context.user
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations.py:1340]: I think this part deserves some comments explaining the business logic here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -142,8 +142,10 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add `page_type_id` when it's possible for `AttributeValue` translations webhook. - #7825 by @fowczarek
 - Optimize available quantity loader. - #7802 by @fowczarek
 - Configure a periodic task for removing empty allocations - #7885 by @fowczarek
-
 - Add webhooks for stock changes: `PRODUCT_VARIANT_OUT_OF_STOCK` and `PRODUCT_VARIANT_BACK_IN_STOCK`  - #7590 by @mstrumeck
+- Allow impersonating user by an app/staff - #7754 by @korycins:
+  - Add `customerId` to `checkoutCustomerAttach` mutation
+  - Add new permision `IMPERSONATE_USER`
```
[the remark this answers is not in the document]
Comment (korycins) [reply CHANGELOG.md:148]: It is something which I also port to 3.0

File: saleor/graphql/checkout/mutations.py [hunk 4]
```diff
@@ -621,6 +623,14 @@ class Arguments:
                 f"The ID of the checkout. {DEPRECATED_IN_3X_INPUT} Use token instead."
             ),
         )
+        customer_id = graphene.ID(
+            required=False,
+            description=(
+                "ID of customer to attach to checkout. Can be used to "
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/checkout/mutations.py:630]: it is something which I also port to 3.0

**Review by eCzerniak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**